src/Symfony/Component/Mailer/Bridge/Amazon/Credential/InstanceCredentialProvider.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Bridge\Amazon\Credential;
+
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mailer\Exception\RuntimeException;
+
+/**
+ * Based on: aws-sdk-php / Credentials/InstanceProfileProvider.php.
+ *
+ * @author Karoly Gossler <connor@connor.hu>
+ */
+class InstanceCredentialProvider
+{
+    const SERVER_URI = 'http://169.254.169.254';
+    const CREDENTIAL_PATH = '/latest/meta-data/iam/security-credentials/';
+
+    public function __construct(int $retries = 3)
+    {
+        $this->retries = $retries;
+        $this->client = HttpClient::create();
+    }
+
+    protected function getInstanceMetadataServerURL(string $roleName): string
+    {
+        return self::SERVER_URI.self::CREDENTIAL_PATH.$roleName;
```

Comment:
I would prefer a single constant with a default value of `http://169.254.169.254/latest/meta-data/iam/security-credentials/%role_name%` and use `str_replace` here (for consistency with what we have already done elsewhere in the component).

Reply:
I would also inline the code as this method is only used once.